hbase-hadoop-compat/src/main/java/org/apache/hadoop/hbase/regionserver/wal/MetricsWALSourceImpl.java

```diff
@@ -88,7 +88,7 @@ public void incrementAppendSize(TableName tableName, long size) {
     if (tableAppendSizeCounter == null) {
       // Ideally putIfAbsent is atomic and we don't need a branch check but we still do it to avoid
       // expensive string construction for every append.
-      String metricsKey = String.format("%s.%s", tableName, APPEND_SIZE);
+      String metricsKey = String.format("%s.%s", tableName.getMetricPrefixTableName(), APPEND_SIZE);
```

Comment:
So the problem here is that, if we have ':' in th table name, it can not be displayed in jmx?

But this is still a breaking change... do we have other ways to obtain the metrics beside jmx? Do these ways also require that there is no ':' in the metrics name?